CMake_Win.cmake

```diff
@@ -34,13 +32,8 @@ foreach (A_DLL_FILE ${DLL_FILES})
     configure_file("${A_DLL_FILE}" "${CMAKE_BINARY_DIR}/${TEMP_VAR_HOLDING_DLL_FILENAME}" COPYONLY)
 endforeach ()
 
-file(GLOB DLL_FILES LIST_DIRECTORIES false "${CMAKE_SOURCE_DIR}/winlibs/lib/xapian/rel/*.dll")
-foreach (A_DLL_FILE ${DLL_FILES})
-    get_filename_component(TEMP_VAR_HOLDING_DLL_FILENAME ${A_DLL_FILE} NAME)
-    configure_file("${A_DLL_FILE}" "${CMAKE_BINARY_DIR}/${TEMP_VAR_HOLDING_DLL_FILENAME}" COPYONLY)
-endforeach ()
-
-file(GLOB DLL_FILES LIST_DIRECTORIES false "${CMAKE_SOURCE_DIR}/winlibs/lib/*.dll")
+set(THIRD_PARTY_DLL "$<IF:$<CONFIG:Debug>,${CMAKE_SOURCE_DIR}/winlibs/lib/dbg/*.dll,${CMAKE_SOURCE_DIR}/winlibs/lib/*.dll>")
+file(GLOB DLL_FILES LIST_DIRECTORIES false "${THIRD_PARTY_DLL}")
 foreach (A_DLL_FILE ${DLL_FILES})
     get_filename_component(TEMP_VAR_HOLDING_DLL_FILENAME ${A_DLL_FILE} NAME)
     configure_file("${A_DLL_FILE}" "${CMAKE_BINARY_DIR}/${TEMP_VAR_HOLDING_DLL_FILENAME}" COPYONLY)
```

Comment:
this part seems does not work , how to make this work?  @shenlebantongying .

Reply:
How about not using fancy features like `#<IF>`

Just have two lists

```
if(CMAKE_BUILD_TYPE MATCHES DEBUG)
  file(GLOB DILL_FILES LIST_DIRECTORIES false "${CMAKE_SOURCE_DIR}/winlibs/lib/dbg/*.dll")
else()
  file(GLOB DILL_FILES LIST_DIRECTORIES false "${CMAKE_SOURCE_DIR}/winlibs/lib/*.dll")
endif()

foreach (A_DLL_FILE ${DLL_FILES})


```

Reply:
ok I will try that.  

On normal cmake build ,will the dll be copied to the target folder.

Reply:
I think the problem is this:

> Since generator expressions are evaluated during generation of the buildsystem, and not during processing of CMakeLists.txt files 

https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#debugging

---

CMake is not a build tool, but rather a build generator.

Steps: `(1.)` Execute the CMake script -> `(2.)` Generate a build system's script like GNU make or ninja or VS Project -> `(3.)` Actually calling compiler & linker etc.

The "cmake-generator-expression (aka `#<IF>`)" happens between `(1.)`->`(2.)`

And the line you mentioned entirely happens on `(1.)` thus it is not expanded, but we want the expanded result (DLL lists) in the following lines (during the execution of cmake script), so `#<>` is unusable.

(I could be wrong, CMake's doc is insanely obscure😅)

Reply:
I have tried your method.it worked.

then how to copy eb.dll to the target folder as well.

Reply:
Maybe this?
```
    target_link_libraries(${GOLDENDICT} PRIVATE eb)
    set_target_properties(eb PROPERTIES
            RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_BINARY_DIR}
            RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_BINARY_DIR}
            )
```